File: packages/ds-lit/src/components/Status/Status.ts
// @generated:start imports
import { LitElement, html, css, nothing } from 'lit';
import { property } from 'lit/decorators.js';
// @generated:end

// @custom:start imports

// @custom:end

// @generated:start types
export type StatusIntent = "info" | "success" | "warning" | "danger" | "error";
// @generated:end

// @custom:start types

// @custom:end

// @generated:start component
export class StatusElement extends LitElement {
  static override styles = css`
    :host { display: contents; }
    .status {
      --fsds-box-model-padding: 0;
      --fsds-box-model-padding-block: 0;
      --fsds-box-model-padding-block-start: var(--fsds-semantic-glyph-badge-size-md-paddingY, 2px);
      --fsds-box-model-padding-block-end: var(--fsds-semantic-glyph-badge-size-md-paddingY, 2px);
      --fsds-box-model-padding-inline: 0;
      --fsds-box-model-padding-inline-start: var(--fsds-semantic-glyph-badge-size-md-paddingX, 8px);
      --fsds-box-model-padding-inline-end: var(--fsds-semantic-glyph-badge-size-md-paddingX, 8px);
      --fsds-box-model-gap: var(--fsds-semantic-glyph-badge-size-md-gap, 4px);
      --fsds-box-model-width: auto;
      --fsds-box-model-min-width: 0;
      --fsds-box-model-max-width: none;
      --fsds-box-model-height: auto;
      --fsds-box-model-min-height: 0;
      --fsds-box-model-max-height: none;
      --fsds-status-color-background-default: var(--fsds-semantic-color-background-secondary, #d0d0d0);
      --fsds-status-color-foreground-primary: var(--fsds-semantic-color-foreground-secondary, #5c5b5c);
      --fsds-status-color-border-default: var(--fsds-semantic-color-border-subtle, #d0d0d0);
      --fsds-status-size-radius-default: var(--fsds-semantic-shape-control-radius-pill, 9999px);
      --fsds-status-size-minHeight: var(--fsds-semantic-glyph-badge-size-md-minHeight, 24px);
      --fsds-status-size-fontSize: var(--fsds-semantic-glyph-badge-size-md-fontSize, 12px);
      --fsds-status-size-border-default: var(--fsds-semantic-shape-control-border-defaultWidth, 1px);
      --fsds-status-text-weight: var(--fsds-semantic-typography-font-weight-medium, 500);
      --fsds-status-typography-lineHeight: var(--fsds-semantic-typography-line-height-collapse, 1);
    }

    .status--info {
      --fsds-status-color-background-default: var(--fsds-semantic-color-background-info-subtle, #95dafb);
      --fsds-status-color-foreground-primary: var(--fsds-semantic-color-foreground-on-info-subtle, #013ab0);
      --fsds-status-color-border-default: var(--fsds-semantic-color-border-info, #034fd6);
    }

    .status--success {
      --fsds-status-color-background-default: var(--fsds-semantic-color-background-success-subtle, #b3dba7);
      --fsds-status-color-foreground-primary: var(--fsds-semantic-color-foreground-on-success-subtle, #2c4f09);
      --fsds-status-color-border-default: var(--fsds-semantic-color-border-success, #3a6614);
    }

    .status--warning {
      --fsds-status-color-background-default: var(--fsds-semantic-color-background-warning-subtle, #fdc67f);
      --fsds-status-color-foreground-primary: var(--fsds-semantic-color-foreground-on-warning-subtle, #6c3a00);
      --fsds-status-color-border-default: var(--fsds-semantic-color-border-warning, #8b4b00);
    }

    .status--danger {
      --fsds-status-color-background-default: var(--fsds-semantic-color-background-danger-subtle, #fac2c2);
      --fsds-status-color-foreground-primary: var(--fsds-semantic-color-foreground-on-danger-subtle, #900909);
      --fsds-status-color-border-default: var(--fsds-semantic-color-border-danger, #b31b1b);
    }

    .status--error {
      --fsds-status-color-background-default: var(--fsds-semantic-color-background-danger-subtle, #fac2c2);
      --fsds-status-color-foreground-primary: var(--fsds-semantic-color-foreground-on-danger-subtle, #900909);
      --fsds-status-color-border-default: var(--fsds-semantic-color-border-danger, #b31b1b);
    }

    .status {
      padding-block-start: var(--fsds-box-model-padding-block-start);
      padding-block-end: var(--fsds-box-model-padding-block-end);
      padding-inline-start: var(--fsds-box-model-padding-inline-start);
      padding-inline-end: var(--fsds-box-model-padding-inline-end);
      gap: var(--fsds-box-model-gap);
      width: fit-content;
      min-width: var(--fsds-box-model-min-width);
      max-width: var(--fsds-box-model-max-width);
      height: var(--fsds-box-model-height);
      min-height: var(--fsds-status-size-minHeight, 24px);
      max-height: var(--fsds-box-model-max-height);
      display: inline-flex;
      align-items: center;
      justify-content: center;
      box-sizing: border-box;
      white-space: nowrap;
      border-style: solid;
      background-color: var(--fsds-status-color-background-default, #d0d0d0);
      color: var(--fsds-status-color-foreground-primary, #5c5b5c);
      border-color: var(--fsds-status-color-border-default, #d0d0d0);
      border-width: var(--fsds-status-size-border-default, 1px);
      border-radius: var(--fsds-status-size-radius-default, 9999px);
      font-size: var(--fsds-status-size-fontSize, 12px);
      font-weight: var(--fsds-status-text-weight, 500);
      line-height: var(--fsds-status-typography-lineHeight, 1);
    }

    .status__icon {
      display: inline-flex;
      align-items: center;
      flex-shrink: 0;
      width: 1em;
      height: 1em;
    }

    .status__label {
      display: inline-block;
      color: var(--fsds-status-color-foreground-primary, #5c5b5c);
    }
  `;

  @property({ type: String }) status!: StatusIntent;

  override connectedCallback(): void {
    super.connectedCallback();
    this.setAttribute("data-fsds-component", "status");
  }

  private computeClasses(): string {
    return [
      "status",
      this.status ? `status--${this.status}` : null,
    ].filter(Boolean).join(" ");
  }

  override render() {
    return html`<span class="${this.computeClasses()}">
  <span class=${'status__icon'} aria-hidden="true"></span>
  <span class=${'status__label'}>
    <slot></slot>
  </span>
</span>`;
  }
}

customElements.define('fsds-status', StatusElement);
// @generated:end

// @custom:start trailing

// @custom:end

File: packages/ds-swiftui/Sources/DsSwiftUI/Components/Status/Status.swift
// @generated:start imports
import SwiftUI
// @generated:end

// @generated:start types
public enum StatusIntent: String, CaseIterable {
    case info
    case success
    case warning
    case danger
    case error
}
// @generated:end

// @generated:start component
/// Token scope data for Status (ir.tokenScopes → RN normal form: data consumed through FsdsTheme at render, never resolved constants). A caseless enum namespace because generic types cannot hold static stored properties.
enum StatusTokens {
    public static let scopes: FsdsComponentTokenScopes = [
        "root": [
            "box-model.padding-block-start": FsdsComponentTokenDefinition(cssVar: "--fsds-box-model-padding-block-start", name: "box-model.padding-block-start", ref: "semantic.glyph.badge.size.md.paddingY", fallback: .string("2px")),
            "box-model.padding-inline-start": FsdsComponentTokenDefinition(cssVar: "--fsds-box-model-padding-inline-start", name: "box-model.padding-inline-start", ref: "semantic.glyph.badge.size.md.paddingX", fallback: .string("8px")),
            "box-model.gap": FsdsComponentTokenDefinition(cssVar: "--fsds-box-model-gap", name: "box-model.gap", ref: "semantic.glyph.badge.size.md.gap", fallback: .string("4px")),
            "box-model.min-height": FsdsComponentTokenDefinition(cssVar: "--fsds-box-model-min-height", name: "box-model.min-height", literal: .string("0")),
            "status.color.background.default": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-background-default", name: "status.color.background.default", ref: "semantic.color.background.secondary", fallback: .adaptive(light: "#f7f7f7", dark: "#313131")),
            "status.color.foreground.primary": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-foreground-primary", name: "status.color.foreground.primary", ref: "semantic.color.foreground.secondary", fallback: .adaptive(light: "#474647", dark: "#a0a0a1")),
            "status.color.border.default": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-border-default", name: "status.color.border.default", ref: "semantic.color.border.subtle", fallback: .adaptive(light: "#d0d0d0", dark: "#474647")),
            "status.size.radius.default": FsdsComponentTokenDefinition(cssVar: "--fsds-status-size-radius-default", name: "status.size.radius.default", ref: "semantic.shape.control.radius.pill", fallback: .string("9999px")),
            "status.size.border.default": FsdsComponentTokenDefinition(cssVar: "--fsds-status-size-border-default", name: "status.size.border.default", ref: "semantic.shape.control.border.defaultWidth", fallback: .string("1px")),
        ],
        "variant_info": [
            "status.color.background.default": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-background-default", name: "status.color.background.default", ref: "semantic.color.background.info.subtle", fallback: .adaptive(light: "#95dafb", dark: "#000a69")),
            "status.color.foreground.primary": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-foreground-primary", name: "status.color.foreground.primary", ref: "semantic.color.foreground.on.info.subtle", fallback: .adaptive(light: "#013ab0", dark: "#00a9fb")),
            "status.color.border.default": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-border-default", name: "status.color.border.default", ref: "semantic.color.border.info", fallback: .adaptive(light: "#034fd6", dark: "#0566fe")),
        ],
        "variant_success": [
            "status.color.background.default": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-background-default", name: "status.color.background.default", ref: "semantic.color.background.success.subtle", fallback: .adaptive(light: "#b3dba7", dark: "#0b2200")),
            "status.color.foreground.primary": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-foreground-primary", name: "status.color.foreground.primary", ref: "semantic.color.foreground.on.success.subtle", fallback: .adaptive(light: "#2c4f09", dark: "#6eb157")),
            "status.color.border.default": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-border-default", name: "status.color.border.default", ref: "semantic.color.border.success", fallback: .adaptive(light: "#3a6614", dark: "#497f21")),
        ],
        "variant_warning": [
            "status.color.background.default": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-background-default", name: "status.color.background.default", ref: "semantic.color.background.warning.subtle", fallback: .adaptive(light: "#fdc67f", dark: "#341400")),
            "status.color.foreground.primary": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-foreground-primary", name: "status.color.foreground.primary", ref: "semantic.color.foreground.on.warning.subtle", fallback: .adaptive(light: "#6c3a00", dark: "#ec8802")),
            "status.color.border.default": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-border-default", name: "status.color.border.default", ref: "semantic.color.border.warning", fallback: .adaptive(light: "#8b4b00", dark: "#ae5d00")),
        ],
        "variant_danger": [
            "status.color.background.default": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-background-default", name: "status.color.background.default", ref: "semantic.color.background.danger.subtle", fallback: .adaptive(light: "#fac2c2", dark: "#440000")),
            "status.color.foreground.primary": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-foreground-primary", name: "status.color.foreground.primary", ref: "semantic.color.foreground.on.danger.subtle", fallback: .adaptive(light: "#900909", dark: "#ee8181")),
            "status.color.border.default": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-border-default", name: "status.color.border.default", ref: "semantic.color.border.danger", fallback: .adaptive(light: "#b31b1b", dark: "#d92d2e")),
        ],
        "variant_error": [
            "status.color.background.default": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-background-default", name: "status.color.background.default", ref: "semantic.color.background.danger.subtle", fallback: .adaptive(light: "#fac2c2", dark: "#440000")),
            "status.color.foreground.primary": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-foreground-primary", name: "status.color.foreground.primary", ref: "semantic.color.foreground.on.danger.subtle", fallback: .adaptive(light: "#900909", dark: "#ee8181")),
            "status.color.border.default": FsdsComponentTokenDefinition(cssVar: "--fsds-status-color-border-default", name: "status.color.border.default", ref: "semantic.color.border.danger", fallback: .adaptive(light: "#b31b1b", dark: "#d92d2e")),
        ],
    ]
}

/// Emitted through the static-content path: passive span root with a single consumer content region.
public struct Status<Content: View>: View {
    private var fsdsScopes: FsdsComponentTokenScopes {
        StatusTokens.scopes
    }
    private let status: StatusIntent?
    private let content: Content
    @Environment(\.fsdsTheme) private var fsdsTheme

    public init(
        status: StatusIntent? = nil,
        @ViewBuilder content: () -> Content
    ) {
        self.status = status
        self.content = content()
    }

    private var layered: [String: FsdsTokenValue?] {
        resolveFsdsLayeredTokens(
            fsdsScopes,
            fsdsTheme,
            layers: ["root", status.map { "variant_\($0.rawValue)" }].compactMap { $0 }
        )
    }

    private func colorSlot(_ suffix: String) -> Color? {
        layered.first { $0.key.hasSuffix(suffix) }?.value?.color
    }

    private func pxSlot(_ suffix: String) -> CGFloat? {
        layered.first { $0.key.hasSuffix(suffix) }?.value?.px
    }

    private var background: Color { colorSlot("color.background.default") ?? .accentColor }
    private var foreground: Color { colorSlot("color.foreground.primary") ?? .primary }
    private var borderColor: Color { colorSlot("color.border.default") ?? .clear }
    private var borderWidth: CGFloat { pxSlot("size.border.default") ?? 0 }
    private var radius: CGFloat { pxSlot("size.radius.default") ?? 0 }
    private var blockPadding: CGFloat { pxSlot("padding-block-start") ?? 0 }
    private var inlinePadding: CGFloat { pxSlot("padding-inline-start") ?? 0 }
    private var gap: CGFloat { pxSlot("box-model.gap") ?? 0 }
    private var minHeight: CGFloat { pxSlot("min-height") ?? 0 }

    public var body: some View {
        content
            .padding(.vertical, blockPadding)
            .padding(.horizontal, inlinePadding)
            .background(background)
            .clipShape(RoundedRectangle(cornerRadius: radius, style: .continuous))
            .foregroundStyle(foreground)
    }
}
// @generated:end
